packages/flutter/lib/src/material/tab_controller.dart

```diff
@@ -205,6 +205,18 @@ class TabController extends ChangeNotifier {
   /// [TabBarView.children]'s length.
   final int length;
 
+  /// The last duration used to change index.
+  ///
+  /// Defaults to [kTabScrollDuration].
+  Duration get lastDuration => _lastDuration;
+  Duration _lastDuration = kTabScrollDuration;
```

Comment:
_lastDuration is not a good way to keep track an on going animation. Maybe use _onGoingAnimationDuration, and the property should be null when there is not animation on going. to prevent the previous animation accidentally leak to the next one

Reply:
Sounds good. Will have a look at this.